scipy/stats/_binned_statistic.py

```diff
@@ -650,6 +635,32 @@ def binned_statistic_dd(sample, values, statistic='mean',
     return BinnedStatisticddResult(result, edges, binnumbers)
 
 
+def _calc_binned_statistic(Vdim, bin_numbers, result, values, stat_func,
+                           is_callable=False):
+    unique_bin_numbers = np.unique(bin_numbers)
+    for vv in builtins.range(Vdim):
+        bin_map = _create_bin_hashmap(bin_numbers, unique_bin_numbers, values, vv)
+        for i in unique_bin_numbers:
+            # if the stat_func is callable, all results should be updated
+            # if the stat_func is np.std, calc std only when binned data is 2
+            # or more for speed up.
+            if is_callable or not (stat_func is np.std and len(bin_map[i]) < 2):
+                result[vv, i] = stat_func(bin_map[i])
+
+
+def _create_bin_hashmap(bin_numbers, unique_bin_numbers, values, vv):
+    """ Create bin hashmap
```

Comment:
Sorry to waffle here but can we remove the hashmap in the ***name*** and change the  docstring to refer say something like "Create hashmap of bin ids to values in bins"? 

Reason being: Hungarian notation (types in names) isn't a good idea for duck typed languages. I realize this is a private function so this is primarily our future selves we are being kind to by choosing descriptive names. 

Pluse if we decide in the future to change the implementation from hashmap to some other data structure, the name would need to change if it contains hashmap. If we remove the implementation becomes something independent of the name. Hope that helps clarify.

Reply:
OK. I changed it to `_create_binned_data`.